### DISCUSSION TITLE
Timestamp

### DISCUSSION THREAD

**Discussion by garyrob** [Chia-Network/chia-blockchain#1748] [Q&A]
There are papers describing the fact that "miners can backdate data using the timestamp on a block" on Ethereum.[1] How reliable are timestamps on the Chia network? It seems like it could be a selling point for Chia if they're more trustworthy.

[1] E.g. https://www.x-mol.com/paper/1347335511108374528

**Comment by hoffmang9** [comment]
Timestamps are far more trustworthy with Chia because of the interspersed proofs of time. You can have someone trying to inject bad timestamps but consensus rules limit that and by simply zooming out a little you can see on chain what the current epoch iterations per second from the fastest Timelord is and verify the sanity of each timestamp as a delta between them.

**Reply by garyrob** [reply]
Thank for answering so quickly. Are you able to supply an estimate of the number of seconds of variance for a truly trustworthy timestamp on Chia?

**Reply by bramcohen** [reply]
The timestamping problems in that paper are over extremely short periods, in the tens of seconds, and it would be ridiculous to rely on that kind of time precision in blockchains in general. You can have a period of no blocks over a minute long just by happenstance. For what it's worth in Chia the timestamp on a new transaction block must be strictly greater than the previous one. There's a bit of a tendency in Solidity to make ridiculous unfulfillable promises and then claim there are security problems when those promises aren't met. Using Uniswap as a price oracle is a far worse sin in this regard.

Over longer time periods (as in, the ones which work) there are tradeoffs between using height and timestamps. Timestamps are fairly precise (within a few minutes even when some farmers are messing with you and even when they're set years in advance) but lack the security lock-in of height, while height has long-term correlated errors with changes to netspace and timelord speed (although not all that much due to the short work difficulty reset interval) and statistical drift correlated with the square root of the number of blocks, which amounts to a standard deviation of about half a day over the course of a year.

**Reply by garyrob** [reply]
Hi Bram, thank you. I'm looking for a new project, and one thing I'm considering would involve timestamping stock market predictions as a piece of it, hence my question. Doesn't sound like Chia's innate timestamping is relevant to that. But now that I'm investigating Chia, other possibilities may come up. If you or anyone has any thoughts to share about things that relate to my background (https://profile.typepad.com/garyrob) feel encouraged to let me know.

**Reply by andri228** [reply]
rly . thx)
